js/ui/map.js

```diff
@@ -249,6 +253,30 @@ class Map extends Camera {
      */
     removeControl(control) {
         control.onRemove(this);
+        if (this._controls[control.constructor.name]) {
+            // delete this._controls[control.constructor.name];
+        }
+        return this;
+    }
+
+
+    /**
+     * Changes position of the control
+     *
+     * @param {string | IControl} control The [`IControl`](#IControl) or the name of the
+     * [`IControl`](#IControl) type you want to move.
+     * @param {string} position on the map to which the control will be moved.
+     * Valid values are `'top-left'`, `'top-right'`, `'bottom-left'`, and `'bottom-right'`.
+     * @returns {Map} `this`
+     * @example
+     * map.setControlPosition('LogoControl', 'top-right');
+     *
+     */
+
+    setControlPosition(control, position) {
+        const _control = typeof control === "string" ? this._controls[control] : control;
+        this.removeControl(_control);
+        this.addControl(_control, position);
```

Comment:
We do have a constructor option for the wordmark, but there's no way currently to change the position of the control at runtime. Our solution to that for other controls is adding the control manually and saving a reference to it so you can add/remove to change the position when you want. This need comes up single page apps that change layout based on user interaction / window-size. If we don't want to support this though that's fine.